### PR TITLE
Update flags documentation with fixes

### PR DESCRIPTION
**Pull request by manshreck**
* Add custom flags docs
* Fix links

### CODE REVIEW

File: docs/cpp/guides/flags.md [hunk 1]
```diff
@@ -410,3 +422,131 @@ Abseil flags library to suppress normal error signaling that occurs when
...
+
+// AbslUnparseFlag converts from an OutputMode to a string.
+// Must be in same namespace as OutputMode.
+
+// Returns a textual flag value corresponding to the OutputMode `mode`.
+std::string AbslUnparseFlag(OutputMode mode) {
+  switch (mode) {
+   case kPlainText: return "plaintext";
+   case kHtml: return "html";
+   default: return SimpleItoa(mode);
```
Comment (tituswinters) [inline docs/cpp/guides/flags.md]: s/SimpleItoa/absl::StrCat/ ?

File: docs/cpp/guides/flags.md [hunk 2]
```diff
@@ -377,7 +387,9 @@ remove reference to the flag from configuration files and startup scripts. Once
 all jobs are starting up without logging warnings about reference to the retired
 flag, the retired flag can be removed completely.
 
-## Special Usage Flags
+For more details, see the [Tip of the Week on retired flags][retired-flags].
```
Comment (tituswinters) [inline docs/cpp/guides/flags.md]: Maybe just abseil.io/tips/90 instead of indirecting the link?

File: docs/cpp/guides/flags.md (hunk 1 above)
Comment (manshreck) [reply docs/cpp/guides/flags.md]: Fixed in source. Will push at next doc push.

File: docs/cpp/guides/flags.md (hunk 2 above)
Comment (manshreck) [reply docs/cpp/guides/flags.md]: I dunno. I prefer this rather than naked links

**Pull request merged**